### ISSUE TITLE
Context menu option is selected automatically when opened in the bottom right corner

### ISSUE THREAD

**Issue by ilmai** [slint-ui/slint#11824]
### Bug Description

When opening a context menu in the bottom right corner, an option is automatically selected.

Looks like this happens because:
- When opening in the corner, the context menu is placed so that an option is under the mouse cursor (this doesn't happen at either the bottom or right edges, only in the corner)
- Menu options can be selected with a right click

IMO a reasonable fix would be to only allow selecting options with a left click

### Reproducible Code (if applicable)

```slint

```

### Environment Details

- Slint Version: 1.16.1
- Platform/OS: MacOS
- Programming Language: Rust
- Backend/Renderer: Custom / Skia

### Product Impact

Commercial audio plugin. This was reported by a user.

**Comment by ogoffart** [comment]
Thanks for the bug report. The issue is that when the context menu would extend beyond the window boundary, it gets repositioned to stay visible (related to #2375 and #4870).
When this  repositioning moves the menu so that an item ends up under the cursor, releasing the right mouse button activates that item immediately.                                                    
                                                                                                                                          
A possible fix would be to add a short delay after the menu opens during which mouse-up events are ignored. 
This way the normal interaction (hold right-click, drag to an item, release) still works, but a quick release right after opening doesn't accidentally activate an item that was placed under the cursor due to repositioning.
